libs/designsystem/src/lib/components/list/list.component.ts

```diff
@@ -159,13 +171,98 @@ export class ListComponent implements OnInit, AfterViewInit, OnChanges {
     }
   }
 
+  private _headerFn(item: any, index: number, items: any[]) {
+    return this.headerTemplate && index === 0 ? true : null;
+  }
+
+  headerFn = this._headerFn.bind(this);
+
+  private _sectionHeaderFn = (item: any, index: number, items: any[]) => {
+    return this.sectionNameMap.get(index);
+  };
+
+  sectionHeaderFn: (
+    item: any,
+    index: number,
+    items: any[]
+  ) => string | null = this._sectionHeaderFn.bind(this);
+
+  private _footerFn(item: any, index: number, items: any[]) {
+    return this.footerTemplate && items && items.length > 0 && items.length - 1 === index
+      ? true
+      : null;
+  }
+
+  footerFn = this._footerFn.bind(this);
+
+  private kirbyItemHeightFn(item: any, index: number) {
+    const dividerHeight = 1;
+    if (this.isFirst(item) || this.isLast(item)) {
+      const edgeHeight =
+        +DesignTokenHelper.itemHeight().split('px')[0] + +DesignTokenHelper.size('xxs')[0];
+      return this.showDivider ? edgeHeight + dividerHeight : edgeHeight;
+    }
+
+    const itemHeight = +DesignTokenHelper.itemHeight().split('px')[0];
+    return this.showDivider ? itemHeight + dividerHeight : itemHeight;
+  }
+
+  isFirst(item: any) {
+    return this.orderMap.get(item) && this.orderMap.get(item).isFirst;
+  }
+  isLast(item: any) {
+    return this.orderMap.get(item) && this.orderMap.get(item).isLast;
+  }
+
   ngOnChanges(): void {
     this.isSectionsEnabled = !!this.getSectionName;
     if (this.isSectionsEnabled && this.items) {
       this.groupedItems = this.groupBy.transform(this.items, this.getSectionName);
+      this.sortedItems = this.groupedItems.reduce((prev, cur) => [...prev, ...cur.items], []);
+      this.sectionNameMap = this.createSectionNameMap();
     } else {
       this.groupedItems = null;
     }
+    this.orderMap = this.createOrderMap();
+  }
+
+  private createSectionNameMap() {
+    const sectionNameMap = new Map<number, string>();
+    const sectionNameSet = new Set<string>();
+    // calculate section name for each item and add them to a index, section name | null map,
+    this.sortedItems.forEach((item: any, index) => {
+      const sectionName = this.getSectionName(item);
+      if (sectionNameSet.has(sectionName)) {
+        // if sectionname is already added to map, add null for other items
+        sectionNameMap.set(index, null);
```

Comment:
We only want the section to be set for the first item, that has a given section, everything else should be null/not show section.

Reply:
Why?

Reply:
Otherwise, it will show the same section multiple times because of the sectionHeaderfn:
`  private _sectionHeaderFn = (item: any, index: number, items: any[]) => {
    return this.sectionNameMap.get(index);
  };`